services/src/main/java/org/keycloak/truststore/FileTruststoreProviderFactory.java

```diff
@@ -106,24 +112,94 @@ public void init(Config.Scope config) {
             }
         }
 
+        if (includeDefault) {
+            truststore = ensureTrustStoreIsWritable(storepath, truststore);
+            File defaultTrustStore = getDefaultTrustStoreFile();
+            if (defaultTrustStore.exists()) {
+                String trustStoreType = System.getProperty("javax.net.ssl.trustStoreType", KeyStore.getDefaultType());
+                String path = defaultTrustStore.getAbsolutePath();
+                mergeTrustStore(truststore, path, loadStore(path, trustStoreType, null));
+            } else {
+                log.warnf("Default truststore was to be included, but could not be found at: %s", defaultTrustStore, type);
+            }
+        }
+
+        if (additionalFiles != null && additionalFiles.length > 0) {
+            truststore = ensureTrustStoreIsWritable(storepath, truststore);
+            for (String file : additionalFiles) {
+                if (file.endsWith(".p12") || file.endsWith(".pfx")) {
+                    mergeTrustStore(truststore, file, loadStore(file, PKCS12, null));
+                } else {
+                    mergePem(truststore, file);
+                }
+            }
```

Comment:
What are the performance implications since it runs at every start of the server?

Reply:
With some very small microbenchmarking the time to process the default java truststore and add 2 PEM certs from the same file was about 2.5 milliseconds on my system (solid state drive) - this overhead is similar to what the fabric8 client will incur on initial load, which quarkus is currently managing).  It looks like each additional PEM cert would be pretty neglible, each additional file will probably add close to a millisecond.  From the issues we currently have we'd potentially be creating a merge of the system default, a user specified file, and the kubernetes certs.

Reply:
I should ask, what generally would you want to see for performance analysis.

Reply:
I'll analyze the performance around it, but I don't expect any major drawbacks on that.

Reply:
@vmuzikar As @shawkins pointed out, I've achieved similar results with two additional PEM and one PKCS12(.p12) certs. The merging of those marked 2 CPU samples in total (sampling rate 1ms; in total 2ms). The consumed time is really negligible in this case.